### PR TITLE
1842: New xlsx option for ignoring certain nodes for improved performance

### PR DESCRIPTION
**Pull request by hofnarwillie**
## Summary

This PR is to mitigate the issue described here: https://github.com/exceljs/exceljs/issues/1842
When the xlsx document contains a single dataValidation applied to the entire worksheet (or a very large range) using a cell address like `A1:AAA1048000` then the `DataValidationsXform.parseClose()` function loops through each cell in the range and unless you have a super computer it runs out of memory. In some cases these validations are not relevant to the context in which the file is being parsed, so a simple performance improvement is to skip over specific XLSX nodes (in this case dataValidations).

Some unrelated linting issues were fixed automatically by the husky pre-commit. I've left them in this PR.

## Test plan

Included a couple of integration tests and ensured all the other tests still pass. Also updated the README to describe the new option.

## Related to source code (for typings update)

Added typescript typings for `options` passed into `workbook.xlsx.readFile()` and `workbook.xlsx.load()`. I have not made any changes to the `stream.xslx.*` interfaces. Would appreciate some input here from the maintainers of the package as I don't know if it is necessary (i.e. if the stream based implementations will use the same options and parsing techniques).

### CODE REVIEW

**Comment by aonamrata** [comment]
can someone from the maintainers group review this?
@guyonroche , @alubbe or @Siemienik

**Comment by Siemienik** [comment]
I would like, however currently I'm overloaded

File: lib/xlsx/xform/sheet/worksheet-xform.js [hunk 1]
```diff
@@ -368,9 +369,11 @@ class WorkSheetXform extends BaseXform {
...
     }
 
-    this.parser = this.map[node.name];
-    if (this.parser) {
-      this.parser.parseOpen(node);
+    if (!this.ignoreNodes.includes(node.name)) {
+      this.parser = this.map[node.name];
+      if (this.parser) {
+        this.parser.parseOpen(node);
+      }
```
Comment (Siemienik) [inline lib/xlsx/xform/sheet/worksheet-xform.js:376]:
```suggestion
    if (this.map[node.name] && !this.ignoreNodes.includes(node.name)) {
      this.parser = this.map[node.name];
      this.parser.parseOpen(node);
```

Comment (Siemienik) [inline lib/xlsx/xform/sheet/worksheet-xform.js:376]:
```suggestion
    if (this.map[node.name] && !this.ignoreNodes.includes(node.name)) {
      this.parser = this.map[node.name];
      this.parser.parseOpen(node);
```

I would like to propose that on purpose to reduce nested ifs

File: spec/integration/data/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!*.xlsx
```
Comment (Siemienik) [inline spec/integration/data/.gitignore:1]:
```suggestion
!*.xlsx

```

File: README.md [hunk 3]
```diff
@@ -2155,6 +2155,12 @@ faster or more resilient.
 
 #### Reading XLSX[⬆](#contents)<!-- Link generated with jump2header -->
 
+Options supported when reading CSV files.
+
+| Field            |  Required   |    Type     |Description  |
+| ---------------- | ----------- | ----------- | ----------- |
+| ignoreNodes      |     N       |  Array      | A list of node names to ignore while loading the XLSX document. Improves performance in some situations. |
```
Comment (Siemienik) [inline README.md:2162]: I would like to propose add available values here:
```suggestion
| ignoreNodes      |     N       |  Array      | A list of node names to ignore while loading the XLSX document. Improves performance in some situations. <br/> Available: `sheetPr`, `dimension`, `sheetViews `, `sheetFormatPr`, `cols `, `sheetData`, `autoFilter `, `mergeCells `, `rowBreaks`, `hyperlinks `, `pageMargins`, `dataValidations`, `pageSetup`, `headerFooter `, `printOptions `, `picture`, `drawing`, `sheetProtection`, `tableParts `, `conditionalFormatting`, `extLst`,|
```

**Review by Siemienik — APPROVED** [review]
Great idea on the topic of performance optimization. Thank you for your contribution! I added some proposals to update Readme and reduce unnecessary if. I hope that's not a problem for you 😄 @hofnarwillie, I appreciate your effort, the tests added, and your code is cool 👍 

@skypesky or @zurmokeeper, would you like to update README_zh?

Will it merge on the next MergeFest
---
This PR was reviewed during the [MergeFest session](https://discord.gg/siema).

**Comment by zurmokeeper** [comment]
> Great idea on the topic of performance optimization. Thank you for your contribution! I added some proposals to update Readme and reduce unnecessary if. I hope that's not a problem for you 😄 @hofnarwillie, I appreciate your effort, the tests added, and your code is cool 👍
> 
> @skypesky or @zurmokeeper, would you like to update README_zh?
> 
> ## Will it merge on the next MergeFest
> This PR was reviewed during the [MergeFest session](https://discord.gg/siema).

Sure, I'm happy to do that, and after this pr is merged into master, I'll raise a new PR to update README_zh.

**Pull request merged**